api/v1alpha1/helmchartproxy_types.go

```diff
@@ -177,6 +188,17 @@ type HelmUninstallOptions struct {
 	Description string `json:"description,omitempty"`
 }
 
+type SecretKeyRef struct {
+	// Name is the name of the Secret containing the OCI credentials.
```

Comment:
```suggestion
		// Secret is a reference to a Secret containing the OCI credentials. If it is not specified, no Secret will be used.
	Secret *corev1.SecretReference `json:"secret"`

	// Key specifies the key in the Secret to use for the OCI credentials. If it is not specified, it will be defaulted to `config.json`.
	// +optional
	Key string `json:"key,omitempty"`
```

I'm thinking we can keep using the original `corev1.SecretReference` type as a best practice so the spec would look something like this:

```
spec:
  credentials:
    secret:
      name: my-secret
      namespace: default
    key: some-path.json
```

Reply:
makes sense